Feature/milestone 2

YoseptF opened the pull request:
Here we made the model, relationships, and views for friendships v1, as well as some integration tests.

The user can now: 
- See pending friendship invitations sent to me from other users
- Accept or reject friendships invitation.
- See a button “send friend request” next to the name of users who are not yet his/her friend. On both users’ list and single-user page.
![image](https://user-images.githubusercontent.com/44252641/77223221-15609c80-6b20-11ea-83d1-2a7ad0c279fb.png)
![image](https://user-images.githubusercontent.com/44252641/77223225-1e516e00-6b20-11ea-8616-8a33f02230e0.png)
![image](https://user-images.githubusercontent.com/44252641/77223231-24dfe580-6b20-11ea-8a39-aa0a9bdce067.png)

YoseptF committed:
we made a ERD

YoseptF committed:
We made an ERD for our project, we also added the relationships right away

YoseptF committed:
We added the 'is_accepted' field in the friendship so that we know if it's pending or accepted

YoseptF committed:
we made some changes so that we were able to work on pg :D

NewIncome committed:
Add the friendship table with some relationships for the user
and the friendship model.
And tried to add some automatic seeds.

YoseptF committed:
we made a couple of assosiations between the user and the friendship using pure SQL queries

NewIncome committed:
Add files and some code needed for rspec testing, up to user model unit tests.

YoseptF committed:
some changes to the view files and logic

YoseptF committed:
we added the invite to friendship button to the list of users and the single user page :D

YoseptF committed:
some styling

NewIncome committed:
Add methods, routes and buttons for accept and reject a user.

YoseptF committed:
We made some integration and model testin with capybara and rspec

mcrd25 reviewed (changes requested):
# Changes Required 🔍 👀 

Hey @YoseptF! Hey @NewIncome 👋 

Great job with this milestone! In fact, everything is meeting requirements except for one tiny issue with timeline posts.

Users should only see posts of people they've confirmed friendship but I can see posts fo users I have not yet confirmed friendship with. Please address and submit another review

See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/auth/sign-in/?next=/library/fast-track-curriculum/69047/path/step/49736080/)

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

YoseptF committed:
Now you can only see your and your friends' posts on the timeline
Also we added a profile link on the navbar, and a little image to see who you are (in case you have a gravatar)

YoseptF committed:
some linter errors fixed

YoseptF commented:
Hi @mcrd25 😃,
We have made the suggested changes:
- Users now can only see their own and their friends' posts
![image](https://user-images.githubusercontent.com/44252641/77265975-d0de1980-6c63-11ea-93c8-eb9254e057e2.png)
![image](https://user-images.githubusercontent.com/44252641/77265985-da678180-6c63-11ea-870d-bfc79459d3aa.png)

---

Also, we added a profile link in the navbar, and a little profile picture (that's if you have a gravatar set up)

![image](https://user-images.githubusercontent.com/44252641/77266059-0d117a00-6c64-11ea-85be-8737e8fc338a.png)

mcrd25 reviewed (approved):
# change Approved 💪 ⏭ 

Hey team!

Good job implementing changes! You may now move unto next milestone
